src/CMakeLists.txt

```diff
@@ -246,17 +230,151 @@ ADD_DEFINITIONS(-DEIGEN_MPL2_ONLY)
 # - external if CERES_DIR_HINTS and FIND_PACKAGE return a valid Ceres setup
 # ==============================================================================
 FIND_PACKAGE(Ceres QUIET HINTS ${CERES_DIR_HINTS})
-IF(Ceres_FOUND)
-   MESSAGE("CERES FOUND")
-ELSE()
-  MESSAGE("CERES NOT FOUND")
+IF(NOT Ceres_FOUND)
   SET(OpenMVG_USE_INTERNAL_CERES ON)
   SET(CERES_INCLUDE_DIRS
     ${CMAKE_CURRENT_SOURCE_DIR}/third_party/ceres-solver/include
     ${CMAKE_CURRENT_SOURCE_DIR}/third_party/ceres-solver/internal/ceres/miniglog
     ${PROJECT_BINARY_DIR}/third_party/ceres-solver/config)
+    FILE(READ "${CMAKE_CURRENT_SOURCE_DIR}/third_party/ceres-solver/VERSION" CERES_CONFIG)
+    STRING(REGEX REPLACE "version ([0-9.]+).*" "\\1" CERES_VERSION ${CERES_CONFIG})
   SET(CERES_LIBRARIES ceres cxsparse)
-ENDIF(Ceres_FOUND)
+ENDIF()
+
+# ==============================================================================
+# Flann
+# ==============================================================================
+# - internal by default (flann),
+# - external if FLANN_DIR_HINTS and a valid Flann setup is found
+# ==============================================================================
+FIND_PATH(FLANN_INCLUDE_DIRS flann.hpp ${FLANN_DIR_HINTS})
```

Comment:
@simogasp , @pmoulon 

I agree with Simone,

* Lets have single FindLibname.cmake as an option for others (that does not have even pkg-config ways), perhaps a nice parametrizable & reusable generic one.

* In our case of Clp, Osi, ClpUtils, Lemon and even for Flann as i've checked now all have official pkg-config handlers so we could exploit cmake internal pkg-config lookup routines.

Lets see Pierre what would do, if he agrees to i refactor this PR asap.